pages/api/sigil/[address]/tokenId/[tokenId].ts

```diff
@@ -0,0 +1,242 @@
+    let non_dao = 0;
+
+    for (let i = 0; i < data.hashes.length; i++) {
+
+      if (data.hashes[i].type === "DAO") {
+        dao += 1;
+      }
+      else {
+        non_dao += 1;
+      }
+    }
+
+    processedWalletData.dao = dao;
+    processedWalletData.non_dao = non_dao;
+    processedWalletData.votes = data.on_chain_votes;
+
+    processedWalletData.proposals = !data.proposals_created ? 0 : data.proposals_created;
+  }
+
+  return processedWalletData;
+}
+
+export default async function handler(req: NextApiRequest, res: NextApiResponse<ResponseMetadata | string>) {
+
+  const { address } = req.query;
+
+  if (typeof(address) !== 'string') {
+    res.status(400).send('address must be a string');
+    return;
+  }
+
+  if (!isValidAddress(address)) {
+    res.status(400).send('valid (non-ens) wallet address must be provided');
+    return;
+  }
+
+  //////////////////////////////////////////
+  //Try to fetch collection JSON data here//
+  //////////////////////////////////////////
+
+  //This can be deleted if try-fetch collection is built - will be redundant
+  if(!Collections.hasOwnProperty(address)) {
+    res.status(400).send('sigil collection either does not exist or has not been integrated yet');
+    return;
+  }
+
+  //So can this probably
+  const collectionJSONData = Collections[address as keyof typeof Collections];
+
+  const sigilContract = getHashesCollectionContract(address);
+
+  const { tokenId } = req.query;
+
+  if (isNaN(Number(tokenId))) {
+    res.status(400).send('tokenId must be a number');
+    return;
+  }
+
+  try {
+    const getTotalSupply = await sigilContract.totalSupply();
+    const rawTotalSupply = getTotalSupply._hex;
+    const totalSupply = parseInt(rawTotalSupply, 16);
+
+    if (Number(tokenId) >= totalSupply) {
+      res.status(400).send('invalid token Id');
+      return;
+    }
+  } catch (error) {
+    console.error(`error fetching total supply: ${error}`);
+  }
+
+  try {
+    const owner = await sigilContract.ownerOf(Number(tokenId));
+    const mintedFilter = sigilContract.filters.Minted();
+    const AllGeneratedEvents = await sigilContract.queryFilter(mintedFilter);
+    const tokenIdEvent = AllGeneratedEvents.find(event => Number(event?.args?.tokenId) === Number(tokenId));
+    const rawHashesTokenId = tokenIdEvent?.args?.hashesTokenId?._hex;
+    const hashesTokenId = parseInt(rawHashesTokenId, 16);
+
+    const tokenData = getTokenAPIData(hashesTokenId);
+    const walletData = getWalletAPIData(owner);
+
+    const parallelDataFetch = await Promise.all([tokenData, walletData]);
+
+    const processedTokenData = processTokenAPIData(parallelDataFetch[0]);
+    const processedWalletData = processWalletAPIData(parallelDataFetch[1]);
+
+    res.status(200).json({
+      name: `${collectionJSONData.name} #${tokenId}`,
+      description: `${collectionJSONData.description}`,
+      image: '',
+      animation_url: `https://sigils.com/media/${address}.html?hash==0x123&attributeB=3`, 
```

Comment:
are these `hash` and `attribute` query params placeholder atm? I assume so, just double-checking

Reply:
Yep, exactly